src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -113,39 +113,45 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod operation)
 
                 writer.Line($"var {message:D} = _pipeline.CreateMessage();");
                 writer.Line($"var {request:D} = {message}.Request;");
-                var method = operation.Request.HttpMethod;
+                var method = clientMethod.Request.HttpMethod;
                 writer.Line($"{request}.Method = {typeof(RequestMethod)}.{method.ToRequestMethodName()};");
 
-                //TODO: Add logic to escape the strings when specified, using Uri.EscapeDataString(value);
-                //TODO: Need proper logic to convert the values to strings. Right now, everything is just using default ToString().
-                //TODO: Need logic to trim duplicate slashes (/) so when combined, you don't end  up with multiple // together
-
                 writer.Line($"var {uri:D} = new RawRequestUriBuilder();");
-                foreach (var segment in operation.Request.HostSegments)
-                {
-                    WriteUriFragment(writer, uri, segment);
-                }
-                writer.RemoveTrailingComma();
-
-                foreach (var segment in operation.Request.PathSegments)
+                foreach (var segment in clientMethod.Request.PathSegments)
                 {
-                    WritePathSegment(writer, uri, segment);
+                    if (!segment.Value.IsConstant && segment.Value.Reference.Name == "nextLink")
+                    {
+                        if (segment.IsRaw)
+                        {
+                            // Artificial nextLink needs additional logic for relative versus absolute links
+                            WritePathSegment(writer, uri, segment, "AppendRawNextLink");
+                        }
+                        else
+                        {
+                            // Natural nextLink parameters need to use a different method to parse path and query elements
+                            WritePathSegment(writer, uri, segment, "AppendRaw");
```

Comment:
No need to use AppendRaw here

Reply:
Discussed offline. Needed to parsing values with both path and query in the same nextLink.